src/protocol/taiko_alethia/DelayedInclusionStore.sol

```diff
@@ -0,0 +1,100 @@
+//// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.28;
+
+import {IBlobRefRegistry} from "../../blobs/IBlobRefRegistry.sol";
+import {IDelayedInclusionStore} from "../IDelayedInclusionStore.sol";
+
+contract DelayedInclusionStore is IDelayedInclusionStore {
+    struct DueInclusion {
+        bytes32 blobRefHash;
+        uint256 due;
+    }
+
+    /// @notice Emitted when a delayed publication is stored
+    /// @param dueInclusion The delayed publication including
+    /// the blob reference hash and due timestamp
+    /// @param sender The address that stored the delayed publication
+    event DelayedInclusionStored(address indexed sender, DueInclusion dueInclusion);
+
+    /// @notice Emitted when a list of delayed inclusions are processed by the inbox
+    /// @param inclusionsList list of inclusions
+    event DelayedInclusionProcessed(Inclusion[] inclusionsList);
+
+    // Append-only queue for delayed inclusions
+    DueInclusion[] private delayedInclusions;
+
+    // Pointer to the first unprocessed element
+    uint256 private head;
+
+    /// @notice The minimum amount of time a delayed publication needs to
+    /// wait in the queue to be included expressed in seconds
+    uint256 public immutable inclusionDelay;
+
+    address public immutable inbox;
+
+    IBlobRefRegistry public immutable blobRefRegistry;
```

Comment:
This should be a virtual function imo, perhaps with a default value of a "canonical" blobregistry:

```solidity
function blobRefRegistry() public virtual returns (IBlobRegistry) {
  return IBlobRegistry(<CANONICAL REGISTRY>);
}
```

This is equivalent to an immutable variable since it's setup as part of the code. Also, enables users to override and bring their own contract if they want.